Promo Tests

jmrcmendoza opened the pull request:
## Description

### Test
Added Controller tests
Added Data-access tests
Added End-points tests
Added GraphQL Tests
Added Model Tests
Added Use Case Tests
Added Entity Test

jmrcmendoza committed:
test: added promo controller test

added insert scenarios
added select scenarios
added update scenarios
added delete scenarios

jmrcmendoza committed:
test: added promo data access test

added insert scenarios

jmrcmendoza committed:
test: updated promo data access test

added select scenarios

jmrcmendoza committed:
test: updated promo data access

added update scenario

jmrcmendoza committed:
test: updated promo data access

added delete promo scenario

jmrcmendoza committed:
test: added promo end point test

added insert scenarios

jmrcmendoza committed:
test: updated promo data access

added select scenarios

jmrcmendoza committed:
test: updated promo end point

added update scenarios

jmrcmendoza committed:
test: updated promo data access

added delete scenarios

jmrcmendoza committed:
test: added promo model test

added insert scenarios

jmrcmendoza committed:
test: updated promo model

added select scenarios

jmrcmendoza committed:
test: updated promo model test

added update scenarios

jmrcmendoza committed:
test: updated promo model test

added delete scenario

jmrcmendoza committed:
test: updated promo model

added delete scenario

jmrcmendoza committed:
Merge branch 'feature/test-promo' of https://github.com/jmrcmendoza/ramda-excerises into feature/test-promo

jmrcmendoza committed:
test: updated mocha config

jmrcmendoza committed:
test: added promo use case test

added insert scenarios

jmrcmendoza committed:
test: updated promo use case test

added select scenarios

jmrcmendoza committed:
test: updated promo use case test

added update scenarios

jmrcmendoza committed:
test: updated promo use case test

added delete scenarios

jmrcmendoza committed:
test: added promo entity test

jmrcmendoza committed:
test: added promo graphql test

added insert scenarios

jmrcmendoza committed:
test: updated promo graphql

remove unused functions

jmrcmendoza committed:
test: updated promo graphql test

added select scenarios

jmrcmendoza committed:
test: updated promo graphql test

added update scenarios

jmrcmendoza committed:
feat: updated updatePromo mutation

added id argument

jmrcmendoza committed:
test: updated promo graphql test

added delete scenarios

ralphcasipe1 commented on koa_crud/test/unit/entities/promos/promos.spec.ts:
```diff
@@ -0,0 +1,134 @@
+      description: chance.sentence(),
+      requiredMemberFields: [MemberFields.REAL_NAME],
+    };
+    await expect(makePromo(data)).to.eventually.be.fulfilled;
```
Can you add this:
```ts
await expect(makePromo(data)).to.eventually.be.fulfilled.and.be.undefined;
```

ralphcasipe1 commented on koa_crud/test/integration/use-cases/promos/promos.spec.ts:
```diff
@@ -0,0 +1,430 @@
+      const result = await selectPromo(lastPromoId);
+
+      expect(result).to.exist;
+      expect(result).to.be.an('object');
```
Can you add another assertion where it compares the `result.id` and the `lastPromoId`?
I think that would also be helpful

ralphcasipe1 replied:
Nice :heavy_check_mark:

ralphcasipe1 commented on koa_crud/test/integration/models/promo.spec.ts:
```diff
@@ -0,0 +1,165 @@
+    });
+  });
+
+  describe('List Promos', () => {
+    it('should retrieve all promo', async () => {
+      const result = await PromoModel.find({}).lean();
+
+      expect(result).to.be.an('array');
+      expect(result).to.have.length.greaterThan(0);
+    });
+
+    it('should retrieve one promo', async () => {
+      const promo = await PromoModel.findOne({}).lean({ virtuals: true });
+
+      const result = await PromoModel.findById(promo.id);
+
+      expect(result).to.be.an('object');
+      expect(result.id).to.equal(promo.id);
+    });
+  });
```
This should not be tested. It is not worth it. Since the `Model.findById` and `Model.find` is already tested in mongoose library.

I think the most noteworthy when testing the Model is the default value of fields when you create something.

ralphcasipe1 commented on koa_crud/test/integration/models/promo.spec.ts:
```diff
@@ -0,0 +1,165 @@
+    });
+  });
+
+  describe('Delete Promo', () => {
+    it('should delete one promo', async () => {
+      const promo = await PromoModel.findOne({}).lean({ virtuals: true });
+
+      const result = await PromoModel.findByIdAndDelete(promo.id);
+
+      expect(result).to.exist;
+      expect(result).to.be.an('object');
+    });
+  });
```
Same here, This is not worth testing

ralphcasipe1 reviewed (changes requested):
Neat tests and it is well-written. There are some test suite that is not worthy to tests. Please see my requested changes.

jmrcmendoza committed:
test: updated promo model

added assertion for promo defaut status
remove select scenario
remove delet scenario

jmrcmendoza committed:
test: updated promo use case

added id assertion in select one promo

jmrcmendoza committed:
test: updated promo entity

updated perfect scenario assertion

ralphcasipe1 reviewed (approved):
LGTM!
